_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -142,6 +133,73 @@ mfxStatus mfxSchedulerCore::Initialize2(const MFX_SCHEDULER_PARAM2 *pParam)
     return MFX_ERR_NONE;
 }
 
+mfxStatus mfxSchedulerCore::SetThreadNum(const void *pComponent, mfxU32 threadNum)
+{
+    MFX_CHECK(pComponent, MFX_ERR_INVALID_HANDLE);
+
+    // If user requested certain number of threads we follow his choice
+    // doing nothing in this function.
+    if (m_param.params.NumThread)
```

Comment:
that's why we will not listen for the components if user requested certain threads number.